src/fpm.f90

```diff
@@ -256,6 +249,34 @@ subroutine build_model(model, settings, package, error)
     end if
 end subroutine build_model
 
+!> Initialize model compiler flags
+subroutine new_compiler_flags(model,settings)
```

Comment:
Why do we need to split the initialization of the compiler object? There should not be a situation where we end up with a partial compiler object.

Reply:
setting of compiler flags was very scattered across the original code:
https://github.com/fortran-lang/fpm/blob/25724f40d694d167d29bf8d9b33a3eef1914ef7a/src/fpm.f90#L67-L84
https://github.com/fortran-lang/fpm/blob/25724f40d694d167d29bf8d9b33a3eef1914ef7a/src/fpm.f90#L170-L174

Because most metapackages add compiler flags, it's necessary they're all set _before_ we resolve the metapackages into the model, at

https://github.com/perazz/fpm/blob/918336d9a97f89bfb87f5a21ee8fa81ada12632b/src/fpm.f90#L74-L75

I could only move those lines up, but having a subroutine seems to make the code more clear